static/js/copy_and_paste.js

```diff
@@ -89,18 +103,33 @@ function copy_handler() {
             skip_same_td_check = true;
         }
 
+        more_than_two_blocks = check_blocks(start_id,end_id);
+
         // we should let the browser handle the copy-paste entirely on its own
         // (In this case, there is no need for our special copy code)
         if (!skip_same_td_check &&
             startc.parents('.selectable_row>div')[0] === endc.parents('.selectable_row>div')[0]) {
             return;
         }
 
+        if (more_than_two_blocks) {
+            for (row = current_msg_list.get_row(start_id);;
+                row = rows.prev_visible(row)) {
+                if (row.prev().hasClass("message_header")) {
+                    content = $('<div>').text(row.prev().text()
+                                        .replace(/\s+/g, " ")
+                                        .replace(/^\s/, "").replace(/\s$/, ""));
+                    div.append($('<p>').append($('<strong>').text(content.text())));
+                    break;
+                }
+            }
+        }
+
             // Construct a div for what we want to copy (div)
         for (row = current_msg_list.get_row(start_id);
              rows.id(row) <= end_id;
              row = rows.next_visible(row)) {
-             if (row.prev().hasClass("message_header")) {
+             if (row.prev().hasClass("message_header") && more_than_two_blocks) {
```

Comment:
hm, this doesn't seem correct.

Reply:
It will select topic header only if more than two blocks are selected.